### PR TITLE
net: eth: Remove old IPv6 interface identifier address

### PR DESCRIPTION
**Pull request by jukkar**
If we change the ethernet MAC address, then we must also remove
the old IPv6 interface identifier (iid) address from the
interface. Otherwise there might not be enough space in the IPv6
address array for the new iid address and beside the old iid
address is not useful any more after the MAC address is changed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: subsys/net/l2/ethernet/ethernet_mgmt.c [hunk 1]
```diff
@@ -88,6 +88,25 @@ static int ethernet_set_config(u32_t mgmt_request,
...
+		{
+			struct in6_addr iid;
+
+			net_ipv6_addr_create_iid(&iid,
+						 net_if_get_link_addr(iface));
+
+			/* No need to check the return value in this case. It
+			 * is not an error in this case if the address is not
+			 * found.
+			 */
```
Comment (rveerama1) [inline subsys/net/l2/ethernet/ethernet_mgmt.c]: How about a warning or info message?

File: subsys/net/l2/ethernet/ethernet_mgmt.c [hunk 2]
```diff
@@ -88,6 +88,25 @@ static int ethernet_set_config(u32_t mgmt_request,
 			return -EACCES;
 		}
 
+#if defined(CONFIG_NET_IPV6)
```
[the remark this answers is not in the document]
Comment (jukkar) [reply subsys/net/l2/ethernet/ethernet_mgmt.c]: Yes, we can do it like that too, just requires that we create the dummy funcs for those. I can add a patch in this set that creates those dummy ones.

File: subsys/net/l2/ethernet/ethernet_mgmt.c (hunk 1 above)
Comment (rveerama1) [reply subsys/net/l2/ethernet/ethernet_mgmt.c]: ok

**Comment by jukkar** [comment]
Added patch that allows call of various IPv6 specific functions even if IPv6 is disabled.

**Comment by jukkar** [comment]
Fixed sanity error when generating documentation.

**Comment by jukkar** [comment]
Rebasing because of doc warning fixes.

**Comment by jukkar** [comment]
recheck

**Comment by pfalcon** [comment]
Re: "net: if: Allow calling IPv6 specific functions". IMHO, that's going more and more confusing, for someone who would just start to get acquainted with the code. OTOH, I fully understand/agree that it's practical and improves maintainability. Bottom line, that it should be described in some imaginable "Zephyr IP stack developer's primer" ;-).

**Comment by pfalcon** [comment]
> This allows use of IS_ENABLED() macro
in other parts of the system.

Oh, now I got (see) it. The intention is not to call ipv6-specific functions without guards, it's just be able to use `if (IS_ENABLED())` guards in addition to the `#ifdef` ones. Ok. (Once again would mention that we should allow indented `#ifdef`s to alleviate that issue too.)

**Review by pfalcon — APPROVED** [review]

**Pull request merged**